### ISSUE TITLE
Add icons for mcfunction files

### ISSUE THREAD

**Issue by SPGoding** [SpyglassMC/Spyglass#341]
~~Blocked by Microsoft/vscode#14662.~~

Blocked by... not having an icon.

**Comment by SPGoding** [comment]
Blocked by Microsoft/vscode#14662.

**Comment by osfanbuff63** [comment]
Maybe a Minecraft grass block style icon? I wouldn't be able to do it, but just had an idea :P

**Comment by Lorenzo0310200** [comment]
fix?

**Comment by Z0rillac** [comment]
I have these three
![icon1x12](https://user-images.githubusercontent.com/77410912/223761115-de78a6bc-b110-4efa-b23e-4ab0241c9aef.png)
![icon2x12](https://user-images.githubusercontent.com/77410912/223761117-5be1ad8b-26ad-4f0b-87a9-6468da9e62f1.png)
![icon3x12](https://user-images.githubusercontent.com/77410912/223761119-61667097-698f-409d-856c-6b82ffb79e15.png)

preview : 
![Capture d’écran 2023-03-08 161907](https://user-images.githubusercontent.com/77410912/223761755-29bb4bf6-d4fa-42ef-8c6d-97ef9d47ebc6.png)

What do you think ?
[big.zip](https://github.com/SpyglassMC/Spyglass/files/10922417/big.zip)

**Comment by Mqxx** [comment]
> I have these three
> ![icon1x12](https://user-images.githubusercontent.com/77410912/223761115-de78a6bc-b110-4efa-b23e-4ab0241c9aef.png)
> ![icon2x12](https://user-images.githubusercontent.com/77410912/223761117-5be1ad8b-26ad-4f0b-87a9-6468da9e62f1.png)
> ![icon3x12](https://user-images.githubusercontent.com/77410912/223761119-61667097-698f-409d-856c-6b82ffb79e15.png)

What do the individual icons represent?

**Comment by Kesuaheli** [comment]
I use ["Datapack Icons" by SuperAnt](https://marketplace.visualstudio.com/items?itemName=SuperAnt.mc-dp-icons). For functions, it uses an impulse command block for the functions defined in the `load.json` function tag, a repeating command block for the functions defined in the `tick.json` function tag, and a chain command block for all other mcfunctions.
It has many other icons for datapack and resource pack files and folders that fit perfectly.

Also, it's not only made for minecraft files and folder structures, there're actually many other file icon textures in a pixilated style (see overview section). And my favourite is a shellscript `.sh` file, which has a nautilus shell as icon xD

**Comment by tigt** [comment]
I whipped up a minimalist, monochrome icon based on the command block, since that seemed like the most straightforward thing that could possibly work:

<img alt="binary-icon" src="https://github.com/SpyglassMC/Spyglass/assets/8072522/9244bcbc-dd74-4e73-bcd6-dd4e2eb229a6" width="32">

```svg
<svg viewBox="0 0 512 512" fill="currentColor" stroke-width="30" xmlns="http://www.w3.org/2000/svg" shape-rendering="optimizeSpeed">
  <rect width="500" height="500" x="6" y="6" fill="none" stroke="currentColor"/>
  <path d="M99 116 v280l17 17h174l157 -157l-157-157H115z" stroke="currentColor" fill="none"/>
  <rect width="31" height="31" x="177" y="178"/>
  <rect width="31" height="31" x="177" y="241"/>
  <rect width="31" height="31" x="240" y="178"/>
  <rect width="31" height="31" x="240" y="241"/>
  <rect width="31" height="31" x="303" y="241"/>
  <rect width="31" height="31" x="240" y="304"/>
  <rect width="31" height="31" x="177" y="304"/>
</svg>
```

Or if you prefer, a version with value gradation between the “indicator lights”:

<img alt="gradated-icon" src="https://github.com/SpyglassMC/Spyglass/assets/8072522/b1e72fae-ddbd-4e54-88d7-680c81208639" width="32">

```svg
<svg viewBox="0 0 512 512" fill="currentColor" stroke-width="30" xmlns="http://www.w3.org/2000/svg" shape-rendering="optimizeSpeed">
  <rect width="500" height="500" x="6" y="6" fill="none" stroke="currentColor"/>
  <path d="M99 116 v280l17 17h174l157 -157l-157-157H115z" stroke="currentColor" fill="none"/>
  <rect width="31" height="31" x="177" y="178" opacity=".8"/>
  <rect width="31" height="31" x="177" y="241" opacity=".6"/>
  <rect width="31" height="31" x="240" y="178" opacity=".5"/>
  <rect width="31" height="31" x="240" y="241" opacity=".6"/>
  <rect width="31" height="31" x="303" y="241" opacity=".9"/>
  <rect width="31" height="31" x="240" y="304" opacity=".5"/>
  <rect width="31" height="31" x="177" y="304"/>
</svg>
```

The exact coordinates probably will need to be tweaked to show nicely at 16&times;16 pixels, but how do you feel about the concept before that?

**Comment by SPGoding** [comment]
Thanks for all the suggestions. Since there is already a VS Code extension providing icons (https://marketplace.visualstudio.com/items?itemName=SuperAnt.mc-dp-icons), I'm thinking to close this issue as out-of-scope and let other more specialized extensions to do their work.
